clubs/views.py

```diff
@@ -164,19 +167,93 @@ def subscription(self, request, *args, **kwargs):
         Return a list of all students that have subscribed to the club, including their names and emails.
         """
         serializer = SubscribeSerializer(Subscribe.objects.filter(club__code=self.kwargs['code']), many=True)
+
+        # intercept excel format data and format it better
+        if request.accepted_renderer.format == 'xlsx':
+            new_data = []
+            for row in serializer.data:
+                new_data.append(OrderedDict([self._format_cell(k, v) for k, v in row.items()]))
+            new_data = ReturnList(new_data, serializer=serializer)
+            return Response(new_data)
+
         return Response(serializer.data)
 
-    @method_decorator(cache_page(60*5))
+    def _format_cell(self, key, value):
+        """
+        Format a cell in the exported Excel spreadsheet, given (column name, cell value).
+        Returns (new column name, new cell value).
+        """
+        if key in set(['tags', 'target_schools', 'target_majors', 'target_years', 'badges', 'school', 'major']):
```

Comment:
Does this recreate this list for each cell? though performance isn't really an issue here it might be better to make it a global static var for the class

Reply:
Ideally I would actually want to detect the appropriate formatting for each field using `instanceof` and then applying appropriate formatting for that field. If I were to do that, it would also be good to precompute each field once instead of once per cell.

I'll revise this to use set syntax! I don't know if that's actually faster, but it also looks better.

Reply:
I agree with the constants, not sure if Python is smart enough to just optimize that for us.

Reply:
Went ahead and made a very fancy mixin to format XLSX files.